scripts/install_KLU_Sundials.py

```diff
@@ -59,132 +269,49 @@ def download_extract_library(url, download_dir):
 parser = argparse.ArgumentParser(
     description="Download, compile and install Sundials and SuiteSparse."
 )
-parser.add_argument("--install-dir", type=str, default=default_install_dir)
+parser.add_argument(
+    "--force",
+    action="store_true",
+    help="Force installation even if libraries are already found. This will overwrite the pre-existing files.",
+)
+parser.add_argument("--install-dir", type=str, default=DEFAULT_INSTALL_DIR)
 args = parser.parse_args()
 install_dir = (
     args.install_dir
     if os.path.isabs(args.install_dir)
     else os.path.join(pybamm_dir, args.install_dir)
 )
 
-# 1 --- Download SuiteSparse
-suitesparse_version = "6.0.3"
-suitesparse_url = (
-    "https://github.com/DrTimothyAldenDavis/"
-    + f"SuiteSparse/archive/v{suitesparse_version}.tar.gz"
-)
-download_extract_library(suitesparse_url, download_dir)
-
-# The SuiteSparse KLU module has 4 dependencies:
-# - suitesparseconfig
-# - AMD
-# - COLAMD
-# - BTF
-suitesparse_dir = f"SuiteSparse-{suitesparse_version}"
-suitesparse_src = os.path.join(download_dir, suitesparse_dir)
-print("-" * 10, "Building SuiteSparse_config", "-" * 40)
-make_cmd = [
-    "make",
-    "library",
-]
-install_cmd = [
-    "make",
-    f"-j{cpu_count()}",
-    "install",
-]
-print("-" * 10, "Building SuiteSparse", "-" * 40)
-# Set CMAKE_OPTIONS as environment variables to pass to the GNU Make command
-env = os.environ.copy()
-for libdir in ["SuiteSparse_config", "AMD", "COLAMD", "BTF", "KLU"]:
-    build_dir = os.path.join(suitesparse_src, libdir)
-    # We want to ensure that libsuitesparseconfig.dylib is not repeated in
-    # multiple paths at the time of wheel repair. Therefore, it should not be
-    # built with an RPATH since it is copied to the install prefix.
-    if libdir == "SuiteSparse_config":
-        env["CMAKE_OPTIONS"] = f"-DCMAKE_INSTALL_PREFIX={install_dir}"
-    else:
-        # For AMD, COLAMD, BTF and KLU; do not set a BUILD RPATH but use an
-        # INSTALL RPATH in order to ensure that the dynamic libraries are found
-        # at runtime just once. Otherwise, delocate complains about multiple
-        # references to the SuiteSparse_config dynamic library (auditwheel does not).
-        env[
-            "CMAKE_OPTIONS"
-        ] = f"-DCMAKE_INSTALL_PREFIX={install_dir} -DCMAKE_INSTALL_RPATH={install_dir}/lib -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=FALSE -DCMAKE_BUILD_WITH_INSTALL_RPATH=FALSE"
-    subprocess.run(make_cmd, cwd=build_dir, env=env, shell=True, check=True)
-    subprocess.run(install_cmd, cwd=build_dir, check=True)
-
-# 2 --- Download SUNDIALS
-sundials_version = "6.5.0"
-sundials_url = (
-    "https://github.com/LLNL/sundials/"
-    + f"releases/download/v{sundials_version}/sundials-{sundials_version}.tar.gz"
-)
+if args.force:
+    print(
+        "The '--force' option is activated: installation will be forced, ignoring any existing libraries."
+    )
+    sundials_found, suitesparse_found = False, False
```

Comment:
I ran into an issue where:

1. I invoked the script with `python scripts/install_KLU_Sundials.py` and it proceeded to install the components as requested
2. I used `nox -s pybamm-requires -- --force` (with `session.posargs` enabled locally as mentioned above) and it returned a CMake cache error, asking me to re-run CMake from a different target directory.

IMO to avoid such accidental errors on re-runs of the script from re-runs of the script, we should delete these folders and the files under them, but only if trying to install with `--force`:

```
build_sundials/
SuiteSparse-6.0.3/
sundials-6.5.0/
```

because these are the extracted folders from the downloaded tarballs, and they might contain leftover CMake configuration files from a previous installation (especially SuiteSparse). If `--force` is not passed then I guess it makes sense to keep them